Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialTypeAssetCreator.cpp

```diff
@@ -348,45 +348,122 @@ namespace AZ
+                                compatible,
+                                "Connecting MaterialProperty %s to Shader Parameter %s: Property Type '%s' is incompatible "
+                                "with GPU Type '%s'",
+                                propertyDescriptor.GetName().GetCStr(),
+                                paramDesc->m_name.c_str(),
+                                ToString(propertyDescriptor.GetDataType()),
+                                paramDesc->m_typeName.c_str());
+
+                            if (m_asset->m_materialShaderParameterLayout.ConnectParameterToSrg(
+                                    paramDesc, m_materialShaderResourceGroupLayout))
+                            {
+                                // we could connect based on the name, so now check if the types are compatible
+                                [[maybe_unused]] bool srgCompatible =
+                                    m_asset->m_materialShaderParameterLayout.IsPropertyTypeCompatibleWithSrg(
+                                        paramDesc, propertyDescriptor.GetDataType());
+                                AZ_Assert(
+                                    srgCompatible,
+                                    "Connecting MaterialProperty %s with type %s to Shader Resource Group field %s is not supported",
+                                    propertyDescriptor.GetName().GetCStr(),
+                                    ToString(propertyDescriptor.GetDataType()),
+                                    paramDesc->m_name.c_str());
+                            }
+                        }
+                    }
                 }
-                break;
-            case MaterialPropertyDataType::Image:
-                outputId.m_itemIndex = RHI::Handle<uint32_t>{m_materialShaderResourceGroupLayout->FindShaderInputImageIndex(shaderInputName).GetIndex()};
-                if (outputId.m_itemIndex.IsNull())
+            }
+        }
+
+        void MaterialTypeAssetCreator::SetMaterialShaderParameterLayout(MaterialShaderParameterLayout layout)
+        {
+            if (!ValidateIsReady())
+            {
+                return;
+            }
+
+            m_asset->m_materialShaderParameterLayout = AZStd::move(layout);
+        }
+
+        const MaterialShaderParameterLayout MaterialTypeAssetCreator::CreateMaterialShaderParameterLayout()
+        {
+            if (!ValidateIsReady())
+            {
+                return {};
+            }
+            if (!ValidateEndMaterialProperty())
+            {
+                return {};
+            }
+
+            // Create the MaterialParameter-Struct-definition from the properties that have a shader connection
+            MaterialShaderParameterLayout shaderParamsLayout{};
+            // the first entry is always the material type
+            shaderParamsLayout.AddMaterialParameter<uint32_t>("m_materialType", true);
+            // TODO: remove this (debug only)
```

Comment:
Should this be removed before merging?

Reply:
see above: i'm actually checking the Material Instance ID in the shader now. Changed the comment